storage/badger/receipts.go

```diff
@@ -142,8 +143,78 @@ func (r *ExecutionReceipts) Index(blockID, receiptID flow.Identifier) error {
 	})
 }
 
+func (r *ExecutionReceipts) IndexByBlockIDAndExecutionID(blockID, executorID, receiptID flow.Identifier) error {
+	return operation.RetryOnConflict(r.db.Update, func(tx *badger.Txn) error {
+
+		err := operation.IndexExecutionReceiptByBlockIDExecutionID(blockID, executorID, receiptID)(tx)
+		if err == nil {
+			return nil
+		}
+
+		if !errors.Is(err, storage.ErrAlreadyExists) {
+			return err
+		}
+
+		// when trying to index a receipt for a block, and there is already a receipt indexed for this block,
+		// double check if the indexed receipt has the same result.
+		// if the result is the same, we could skip indexing the receipt
+		// if the result is different, then return error
+		var storedReceiptID flow.Identifier
+		err = operation.LookupExecutionReceiptByBlockIDExecutionID(blockID, executorID, &storedReceiptID)(tx)
+		if err != nil {
+			return fmt.Errorf("there is a receipt stored already, but cannot retrieve the ID of it: %w", err)
+		}
+
```

Comment:
`storedReceiptID` could be `nil` at this point, right? Could the function bail out early in that case?

Reply:
wouldn't I get a `storage.ErrNotFound` if `storedReceiptID` couldn't be found?

Reply:
Ah yep my bad!